colcon-test/action.yaml

```diff
@@ -0,0 +1,119 @@
+      shell: bash
+
+    - name: Set up colcon-mixin
+      run: |
+        colcon mixin add default https://raw.githubusercontent.com/colcon/colcon-mixin-repository/master/index.yaml || true
+        colcon mixin update default
+      shell: bash
+
+    - name: Install packages for coverage commands
+      run: |
+        sudo apt-get -yqq update
+        sudo apt-get -yqq install lcov python3-pip
+        pip3 install colcon-lcov-result colcon-coveragepy-result
+      shell: bash
+
+    - name: Clone dependency packages
+      if: ${{ inputs.build-depends-repos != ''}}
+      run: |
+        git config --local --unset-all http.https://github.com/.extraheader || true
+        git config --global url.https://x-access-token:${{ inputs.token }}@github.com.insteadOf 'https://github.com'
+        mkdir dependency_ws
+        vcs import dependency_ws < ${{ inputs.build-depends-repos }}
+      shell: bash
+
+    - name: Run rosdep install
+      run: |
+        sudo apt-get -yqq update
+        rosdep update
+        DEBIAN_FRONTEND=noninteractive rosdep install -yqq --from-paths . --ignore-src --rosdistro ${{ inputs.rosdistro }}
+      shell: bash
+
+    - name: Restore build files from cache
+      id: restore-build-files
+      uses: actions/cache@v2
+      with:
+        path: |
+          ./build
+          ./install
+        key: build-${{ github.sha }}
+
+    - name: Test
+      id: test
+      run: |
+        set -e
+        . /opt/ros/${{ inputs.rosdistro }}/setup.sh
+
+        colcon lcov-result --zero-counters \
+          --verbose \
+          --packages-above ${{ inputs.target-packages }} \
+          --lcov-base lcov_${{ inputs.label }}
+
+        colcon lcov-result --initial \
+          --verbose \
+          --packages-above ${{ inputs.target-packages }} \
+          --lcov-base lcov_${{ inputs.label }}
+
+        colcon test --event-handlers console_cohesion+ \
+          --mixin coverage-pytest \
+          --packages-above ${{ inputs.target-packages }} \
+          --ctest-args -L "${{ inputs.label }}" \
+          --return-code-on-test-failure
+
+        colcon lcov-result \
+          --verbose \
+          --packages-above ${{ inputs.target-packages }} \
+          --lcov-base lcov_${{ inputs.label }} || true
+
+        colcon coveragepy-result \
+          --verbose \
+          --packages-above ${{ inputs.target-packages }} \
+          --coveragepy-base coveragepy_${{ inputs.label }} || true
+
+        coverage_files=()
+        lcov_coverage_file=lcov_${{ inputs.label }}/total_coverage.info
+        if [[ -f $lcov_coverage_file ]]; then
+          coverage_files+=("$lcov_coverage_file")
+        fi
+
+        coveragepy_coverage_file=coveragepy_${{ inputs.label }}/.coverage
+        if [[ -f $coveragepy_coverage_file ]]; then
+        coverage_files+=("$coveragepy_coverage_file")
+        fi
+
+        # Join coverage file names
+        IFS=, eval 'coverage_files_str="${coverage_files[*]}"'
```

Comment:
I remember setting `IFS` may cause some problems because it will remain after this command. I think considering another way is better.

Reply:
If you use IFS inside `$()`, it seems we can avoid the problem.
https://qiita.com/kawaz/items/b82da76ac93b32ddc364#join

Reply:
Good suggestion! ff6b0ee